src/common/AdaptiveStream.cpp

```diff
@@ -549,13 +549,13 @@ bool AdaptiveStream::ensureSegment()
         observer_->OnSegmentChanged(this);
 
       uint32_t nextsegmentPosold = current_rep_->get_segment_pos(nextSegment);
-      uint32_t nextsegno = current_rep_->getSegmentNumber(nextSegment);
+      uint64_t nextsegno = current_rep_->getSegmentNumber(nextSegment);
       AdaptiveTree::Representation* newRep;
       bool lastSeg =
           (current_period_ != tree_.periods_.back() &&
            nextsegmentPosold + available_segment_buffers_ == current_rep_->segments_.size() - 1);
       
-      if (segment_buffers_[0].segment_number == ~0L || valid_segment_buffers_ == 0 ||
+      if (segment_buffers_[0].segment_number == ~0LL || valid_segment_buffers_ == 0 ||
```

Comment:
should be `if (segment_buffers_[0].segment_number == ~0ULL || valid_segment_buffers_ == 0 ||`